lib/PhasarLLVM/TypeHierarchy/DIBasedTypeHierarchy.cpp

```diff
@@ -189,6 +219,86 @@ DIBasedTypeHierarchy::DIBasedTypeHierarchy(const LLVMProjectIRDB &IRDB) {
   buildTypeHierarchy(TG, VertexTypes, TransitiveDerivedIndex, Hierarchy);
 }
 
+static const llvm::DICompositeType *
+stringToDICompositeType(const LLVMProjectIRDB *IRDB,
+                        const llvm::StringRef DITypeName) {
+  llvm::DebugInfoFinder DIF;
+  const auto *Module = IRDB->getModule();
+  DIF.processModule(*Module);
```

Comment:
Maybe refactor the functions `stringToDICompositeType` and `stringToDIType`, such that they take a `llvm::DebugInfoFinder` as parameter to avoid the re-computation of the types in each iteration of the below loops within the `DIBasedTypeHierarchy` ctor